### DISCUSSION TITLE
How to install older version of package when there is a newer version installed?

### DISCUSSION THREAD

**Discussion by arphox** [microsoft/winget-cli#4002] [Q&A] [answered]
Hi!

I have the `.NET SDK 7.0.404` installed, and I want to install the `7.0.100` version too, but I couldn't find a way to do this. How do I do this?

```
> winget list Microsoft.DotNet.SDK.7
Name                             Id                     Version Source
-----------------------------------------------------------------------
Microsoft .NET SDK 7.0.404 (x64) Microsoft.DotNet.SDK.7 7.0.404 winget
```
```
> winget install --id Microsoft.DotNet.SDK.7 --version 7.0.100 --no-upgrade --exact
A package version is already installed. Installation cancelled.
```
☹️

**Comment by Trenly** [comment, marked as answer]
`--no-upgrade` only prevents the install flow from converting to the upgrade flow and upgrading an exising package, which is why its exiting after detecting the package is already installed. You're going to want to use `--force` instead

**Reply by arphox** [reply]
I tried `winget install --id Microsoft.DotNet.SDK.7 --version 7.0.100 --no-upgrade --exact --force`, and it actually installed the requested version, however after installing that, the output of the winget list command is weird:

```
> winget list Microsoft.DotNet.SDK.7
Name                             Id                     Version Available Source
--------------------------------------------------------------------------------
Microsoft .NET SDK 7.0.100 (x64) Microsoft.DotNet.SDK.7 7.0.100 7.0.404   winget
```
(I expect it to show both installed versions separately)

```
> dotnet --list-sdks
6.0.417 [C:\Program Files\dotnet\sdk]
7.0.100 [C:\Program Files\dotnet\sdk]
7.0.404 [C:\Program Files\dotnet\sdk]
8.0.100 [C:\Program Files\dotnet\sdk]
```

So is this another bug, or missing feature, then?

**Reply by Trenly** [reply]
There are several issues with packages that install versions side by side. If you take a look under the https://github.com/microsoft/winget-cli/labels/Side-by-Side label you'll find this is known and being worked on. In fact, theres a specific issue for listing by ID with side-by-side installs.
* https://github.com/microsoft/winget-cli/issues/2726

To get the right output, you would need to list by name - `winget list '.NET SDK 7.0'`

**Reply by arphox** [reply]
Confirmed, `>winget list ".NET SDK 7.0"` works:
```
>winget list ".NET SDK 7.0"
Name                             Id                     Version Available Source
--------------------------------------------------------------------------------
Microsoft .NET SDK 7.0.100 (x64) Microsoft.DotNet.SDK.7 7.0.100 7.0.404   winget
Microsoft .NET SDK 7.0.404 (x64) Microsoft.DotNet.SDK.7 7.0.404           winget
```

Thank you for all your help!
